File: src/main/java/ftnhps/movieenthusiasts/friendships/FriendshipService.java
package ftnhps.movieenthusiasts.friendships;

import java.util.List;

import ftnhps.movieenthusiasts.users.User;

public interface FriendshipService {

	Friendship findOne(Long id);
	
	List<Friendship> findAll();
	
	List<Friendship> findAllOfUser(User user);
	
	Friendship add(Long senderId, Long receiverId);
	
	Friendship accept(Long id, User receiver);
	
	boolean remove(Long id, User user);
}

File: src/main/resources/static/js/core/friendship/friendship.service.js
'use strict';

angular.module('core.friendship')
	.service('FriendshipService', function($http) {
		this.getAll = () => {
			return $http.get('/api/friendships/');
		};
		this.add = (receiverId) => {
			return $http.post(`/api/friendships/${receiverId}`);
		};
		this.accept = (id) => {
			return $http.put(`/api/friendships/${id}`);
		};
		this.remove = (id) => {
			return $http.delete(`/api/friendships/${id}`);
		};
	});
